src/librustc_mir/borrow_check/mod.rs

```diff
@@ -1521,9 +1530,17 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
 
         // check for loan restricting path P being used. Accounts for
         // borrows of P, P.a.b, etc.
-        'next_borrow: for i in flow_state.borrows.elems_incoming() {
-            // FIXME for now, just skip the activation state.
-            if i.is_activation() { continue 'next_borrow; }
+        let mut elems_incoming = flow_state.borrows.elems_incoming();
```

Comment:
Pre-existing, but I feel like the enclosing function here deserves a comment explaining what it does, and I would expect this (quite clever, actually) bit here to be in that comment (i.e., invokes with either a RESERVATION or ACTIVATION for each borrowed path, whichever is more specific).

Reply:
(addressed by new commit injected earlier in commit history)